Cython/Compiler/ExprNodes.py

```diff
@@ -9400,6 +9402,7 @@ def generate_cyfunction_code(self, code):
                 self.get_py_mod_name(code),
                 code_object_result,
                 code.error_goto_if_null(self.result(), self.pos)))
```

Comment:
This still leaks a reference to `temporary_dict` if the constructor call fails and exits here. You need to allocate a new (managed) temp variable for this. See, for example, [here](https://github.com/cython/cython/blob/3dc2b9dfc23638fbef2558d619709b5235d5df08/Cython/Compiler/ExprNodes.py#L7825-L7835).

Reply:
I didn't know that existed! This worked out great.

Reply:
Done! This should now be correct (unless I misused any of the temporary apis)